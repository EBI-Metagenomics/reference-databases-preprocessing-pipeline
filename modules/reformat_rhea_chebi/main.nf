process REFORMAT_RHEA_CHEBI {
    label 'light'
    container 'quay.io/biocontainers/python:3.10'

    input:
    path rhea_rheactions_gz

    output:
    path 'rhea_chebi_mapping_*.tsv', emit: tsv_rhea_chebi_mapping

    script:
    "reformat_rhea_chebi_mapping.py ${rhea_rheactions_gz} rhea_chebi_mapping_${params.rheadb_version}.tsv"
}