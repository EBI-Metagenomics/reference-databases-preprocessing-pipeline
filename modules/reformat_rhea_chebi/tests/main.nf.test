nextflow_process {

    name "Test Process REFORMAT_RHEA_CHEBI_FILE"
    script "../main.nf"
    process "REFORMAT_RHEA_CHEBI_FILE"

    test("Convert test file to TSV format") {

        when {
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/rhea-reactions.txt.gz", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
