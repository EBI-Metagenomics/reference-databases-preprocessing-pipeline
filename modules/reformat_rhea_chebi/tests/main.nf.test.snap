{
    "Convert test file to TSV format": {
        "content": [
            {
                "0": [
                    "rhea_chebi_mapping_135.tsv:md5,509e2f6d4c088073284e6639e71b19f7"
                ],
                "tsv_rhea_chebi_mapping": [
                    "rhea_chebi_mapping_135.tsv:md5,509e2f6d4c088073284e6639e71b19f7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T13:37:25.630037"
    }
}